### ISSUE TITLE
Fix failing jobs

### ISSUE THREAD

**Issue by m4olivei** [Lullabot/mpx-php#180]
About 2 months ago jobs started failing with the following error:

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires cache/cache dev-master#83fabc5bad8b90f7981c97898345df8935b0bf55, found cache/cache[0.1.0, 0.2.0, 0.3, 0.4.0, 1.0.0, 2.0.x-dev] but it does not match the constraint.


Exited with code exit status 2
```

**Comment by deviantintegral** [comment]
Yay, tests!

Boo, do we get emails on failures on the branch?

**Comment by m4olivei** [comment]
Yeah, we do, I was ignoring them. Light sprint this sprint :).

**Comment by m4olivei** [comment]
I'm trying to work out why Composer doesn't want to pull a pinned commit of `master` for this package anymore. It used to without issue. I don't know enough about Composer in this regard though, so I must needs do some research before I get anywhere with this.

**Comment by deviantintegral** [comment]
Perhaps double check that the commit literally exists?

Did cache/cache get a new major version , and now says that "master" is 2.x and not 1.x? Basically composer will load the main / master branch and look at the composer.json to determine what "version" to consider other branches as.

If that's the case, we could probably fix this by doing `dev-master#83fabc5bad8b90f7981c97898345df8935b0bf55 as 1.0`... and now I wonder if that should be an ADR.

**Comment by m4olivei** [comment]
> Perhaps double check that the commit literally exists?

Good thought. It does: https://github.com/php-cache/cache/commit/83fabc5bad8b90f7981c97898345df8935b0bf55

Also, verified that I'm looking at the right repository:

```
$ composer search -f json cache/cache
composer search -f json cache/cache
Info from https://repo.packagist.org: #StandWithUkraine
[
    {
        "name": "cache/cache",
        "description": "Library of all the php-cache adapters",
        "url": "https://packagist.org/packages/cache/cache",
        "repository": "https://github.com/php-cache/cache",
        "downloads": 2014823,
        "favers": 236
    },
    // ...
]
```

> Did cache/cache get a new major version , and now says that "master" is 2.x and not 1.x? Basically composer will load the main / master branch and look at the composer.json to determine what "version" to consider other branches as.

> If that's the case, we could probably fix this by doing dev-master#83fabc5bad8b90f7981c97898345df8935b0bf55 as 1.0... and now I wonder if that should be an ADR.

That worked! But also, huh? The package [hasn't got a new major version](https://github.com/php-cache/issues/issues/150#issuecomment-1120175481), though a few folks are calling for that and it seems poised for that. The [repo doesn't have any 2.x tags created](https://github.com/php-cache/cache/tags). On packagist, a `2.0.x-dev` release is showing. Not clear to me how that got there :thinking:

**Comment by m4olivei** [comment]
Nope, never mind.

https://app.circleci.com/pipelines/github/Lullabot/mpx-php/175/workflows/82890bd7-0844-4534-a309-0ebb6b9dc45e/jobs/2929

**Comment by m4olivei** [comment]
Weird. The package authors must have dropped `dev-master` as a version on Packagist at some point. It's not listed. Fortunately they do list `2.0.x-dev`, which is based on their `2.0` branch, which includes the commit we want, so I changed branch name in the version constraint to `2.0.x-dev`, pinned to the same commit, and away we go.

**Comment by m4olivei** [comment]
Figured out why this happened:

https://github.com/php-cache/issues/issues/150#issuecomment-1128063000

Basically, composer 2.2.0 is more strict with `composer validate` than it was before. It doesn't like the conflicting `conflct` and `replace` keys that `cache/cache` has, and throws errors, which Packagist takes to mean not to offer the branch as a version. So it stopped being offered as a version at some point by Packagist, and should start being offered when fixed.

There is an open PR to fix this already: https://github.com/php-cache/cache/pull/266

I connected some dots on the upstream issues / pull requests that are relevant. Hopefully the authors address it :crossed_fingers:
